docs/transfer/3-5-user-rejects-tpr.puml

```diff
@@ -0,0 +1,64 @@
+@startuml
+
+title Transfer: 3.5 User Rejects Thirdparty Transaction Request
+
+box "PISP"
+participant "PISP Server" as D1
+end box
+box "Mojaloop"
+    participant Switch as S
+end box
+box "DFSP A"
+    participant "DFSP A\n(Payer)" as D2
+end box
+
+
+... PISP has initiated Thirdparty Transaction Request with ""POST /thirdpartyRequests/transactions""...
+
+... DFSP A has received quote, and asked PISP to verify...
+
+note right of D1
+  PISP looks up the ""transactionRequestId"" and
+  checks the quote with the user.
+end note
+
+note right of D1
+  User doesn't wish to proceed with the transaction.
+end note
+
+rnote right of D1 #Light
+**""PUT /thirdpartyRequests/authorizations/33333333-0000-0000-0000-000000000000""**
+""FSPIOP-Source: pispa""
+""FSPIOP-Destination: dfspa""
+{
+  "responseType": "REJECTED"
+}
+end note
+D1 -> S: ""PUT /thirdpartyRequests/authorizations/33333333-0000-0000-0000-000000000000""
+S --> D1: ""200 OK""
+S -> D2: ""PUT /thirdpartyRequests/authorizations""\n""/33333333-0000-0000-0000-000000000000""
+D2 --> S: ""200 OK""
+
+D2 -> D2: Look up the ""transactionRequestId"" for this ""authorizationId""
+D2 -> D2: See that the user rejected the Thirdparty Transaction Request
+
+rnote over D2 #Light
+**""PATCH /thirdpartyRequests/transactions/00000000-0000-0000-0000-000000000000""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "transactionRequestState": "REJECTED",
```

Comment:
`transactionId` is required property for PATCH

```suggestion
  "transactionId": "1111....",
  "transactionRequestState": "REJECTED",
```

Reply:
Yeah, but I think we've just found a case where `transactionId` wouldn't exist yet

Reply:
nope :-D  `transactionId` already exists. 
It has been allocated before the quote request happens (and quote requires it)
PISP is informed about it by receiving PUT with `transactionRequestState='RECEIVED'` a step before.
User accepts/rejects tpr seeing quote details.

Reply:
Ah - of course! That's a happy accident :)

Reply:
I suggest keeping `transactionId` as a required field for PATCH - the client code will be then simpler -> no need to check for the existence of this property before accessing.